[FIX] sale_elaboration: compatible with delivery (don't compute notes)

yajo opened the pull request:
Under certain circumstances, elaboration notes were deleted. For example, when adding or removing an elaboration.

That's more or less OK, but also when delivery is installed along with this module, then the notes removal is unpredictable.

Here I fix https://github.com/OCA/sale-workflow/issues/3154 by removing the computation on notes.

Until now, notes were just repeating the same as we had in elaborations. Since the information was duplicated, the `elaboration_ids` field never appeared in reports.

Now the notes are just a normal text field that does nothing if empty. Since it will be usually empty, now reports need to include `elaboration_ids` too.

@moduon MT-6164

yajo committed:
[FIX] sale_elaboration: compatible with delivery (don't compute notes)

Under certain circumstances, elaboration notes were deleted. For example, when adding or removing an elaboration.

That's more or less OK, but also when delivery is installed along with this module, then the notes removal is unpredictable.

Here I fix https://github.com/OCA/sale-workflow/issues/3154 by removing the computation on notes.

Until now, notes were just repeating the same as we had in elaborations. Since the information was duplicated, the `elaboration_ids` field never appeared in reports.

Now the notes are just a normal text field that does nothing if empty. Since it will be usually empty, now reports need to include `elaboration_ids` too.

@moduon MT-6164

Gelojr reviewed (approved):
Nice @yajo LGTM
https://www.loom.com/share/2dc11399f56349cc95d55ab6f774c215?sid=857b7a20-5c20-49ab-bd6c-b8f28a8ab391

fcvalgar reviewed (approved):
Functional review.
LGTM, thank you @yajo 

[[FIX] sale_elaboration: compatible with delivery (don't compute notes) #3165 - Fran - Watch Video](https://www.loom.com/share/0325d8ef50934896bb8c4b7dbef2321c)

yajo commented:
Thanks @fcvalgar and @Gelojr but there are a couple of missing points to test in your functional reviews, at least in the ones recorded:

1. Check that https://github.com/OCA/sale-workflow/issues/3154 is fixed by making sure the module integrates well with delivery.
2. Check that there are no regressions in `sale_order_product_recommendation_elaboration`.

Could you please update your reviews? Thanks!

yajo committed:
[FIX] sale_order_product_recommendation_elaboration: compatibility with latest upstream changes

WIth https://github.com/OCA/sale-workflow/pull/3165, sale_elaboration won't compute elaboration notes automatically.

Adapting tests here to reflect that new reality.

@moduon MT-6164

yajo commented:
I just pushed a compatibility fix for `sale_order_product_recommendation_elaboration`. In reality it wasn't breaking AFAICS, but the tests were testing it with old assumptions.

Shide commented on sale_elaboration/reports/report_deliveryslip.xml:
```diff
@@ -18,9 +18,9 @@
             position="attributes"
         >
-            <attribute
+            <attributed
```
missclick "d"

yajo replied:
Thanks! fixed

fcvalgar commented:
> Thanks @fcvalgar and @Gelojr but there are a couple of missing points to test in your functional reviews, at least in the ones recorded:
> 
> 1. Check that [[16.0][BUG] sale_elaboration with delivery incompatibility #3154](https://github.com/OCA/sale-workflow/issues/3154) is fixed by making sure the module integrates well with delivery.
> 2. Check that there are no regressions in `sale_order_product_recommendation_elaboration`.
> 
> Could you please update your reviews? Thanks!

My new review 

https://www.loom.com/share/9cc09c3609ec431a842d16bc3b6b45c7?sid=5dc0ecc5-e5d4-4f22-a8fe-c3d855874cab

rafaelbn reviewed (approved):
/ocabot merge patch
